### PR TITLE
papis: 0.5.2 -> 0.5.3

### PR DESCRIPTION
**Pull request by nico202**
###### Motivation for this change
Fixed tests as pointed out in https://github.com/papis/papis/issues/58

### CODE REVIEW

**Review by dotlambda — COMMENTED** [review]
This seems fine so far.
However, looking at https://travis-ci.org/papis/papis/jobs/344071736, I see that we're only running 11/35 tests.
Maybe you could specify
```
checkPhase = ''
  make test
  make test-non-pythonic
'';
```
But then, we'd need to package https://pypi.org/project/dmenu/ and at least add `pytest` to `checkInputs`: https://github.com/papis/papis/blob/master/setup.py#L87

**Comment by nico202** [comment]
@dotlambda done, but there's a failing test on my machine

**Comment by adisbladis** [comment]
@GrahamcOfBorg build papis

File: pkgs/tools/misc/papis/default.nix [hunk 1]
```diff
@@ -1,39 +1,43 @@
...
 }:
 
 buildPythonApplication rec {
   pname = "papis";
-  version = "0.5.2";
+  version = "0.5.3";
 
   # Missing tests on Pypi
   src = fetchFromGitHub {
     owner = "alejandrogallo";
```
Comment (adisbladis) [inline pkgs/tools/misc/papis/default.nix]: Github redirects from `https://github.com/alejandrogallo/papis` to `https://github.com/papis/papis` so this should be updated.

**Comment by adisbladis** [comment]
@nico202 This is because that test is attempting to use networking (which is not available in the build environment).

You can skip that test.

**Review by dotlambda — APPROVED** [review]
Please squash the last two commits and name them something like `papis: fix tests`

**Comment by nico202** [comment]
I'll squash everything as soon as it starts working. I can't figure out the problem now

File: pkgs/tools/misc/papis/default.nix [hunk 2]
```diff
@@ -1,40 +1,52 @@
...
 
+  checkInputs = [ pytest ];
+
   # Papis tries to create the config folder under $HOME during the tests
-  preCheck = ''
+  checkPhase = ''
     mkdir -p check-phase
+    patchShebangs tests
+    sed -i "s|^papis |$out/bin/papis |" tests/bash/test_default.sh
+    sed -i "s|papis |$out/bin/papis |" tests/bash/test_help.sh
```
Comment (dotlambda) [inline pkgs/tools/misc/papis/default.nix]: You can probably do this with
```
export PATH=$out/bin:$PATH
```

File: pkgs/tools/misc/papis/default.nix [hunk 3]
```diff
@@ -1,40 +1,52 @@
...
+    # Still don't know why this fails
+    sed -i 's/--set dir=hello //' tests/bash/test_default.sh
+
+    sed -i 's/test_downloader_getter(self):/disabled_test_downloader_getter(self):/' papis/downloaders/tests/test_main.py
```
Comment (dotlambda) [inline pkgs/tools/misc/papis/default.nix]: Add a comment that this tests requires a network connection

File: pkgs/tools/misc/papis/default.nix [hunk 4]
```diff
@@ -1,40 +1,52 @@
...
     vim
   ];
 
+  checkInputs = [ pytest ];
+
   # Papis tries to create the config folder under $HOME during the tests
-  preCheck = ''
+  checkPhase = ''
     mkdir -p check-phase
+    patchShebangs tests
```
Comment (dotlambda) [inline pkgs/tools/misc/papis/default.nix]: This belongs into `pre/postPatch`

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/papis/default.nix (hunk 2 above)
Comment (nico202) [reply pkgs/tools/misc/papis/default.nix]: better thanks

File: pkgs/tools/misc/papis/default.nix [hunk 5]
```diff
@@ -1,40 +1,53 @@
...
     repo = pname;
     rev = "v${version}";
-    sha256 = "0cw6ajdaknijka3j2bkkkn0bcxqifk825kq0a0rdbbmc6661pgxb";
+    sha256 = "1yc4ilb7bw099pi2vwawyf8mi0n1kp87wgwgwcwc841ibq62q8ic";
   };
 
-  postPatch = "sed -i 's/configparser>=3.0.0/# configparser>=3.0.0/' setup.py";
+  postPatch = ''
+      sed -i 's/configparser>=3.0.0/# configparser>=3.0.0/' setup.py
+      patchShebangs tests
```
Comment (dotlambda) [inline pkgs/tools/misc/papis/default.nix]: indentation

**Comment by dotlambda** [comment]
I'll look at the `--set dir=hello` thing later today.

File: pkgs/tools/misc/papis/default.nix [hunk 6]
```diff
@@ -1,40 +1,53 @@
...
+    export PATH=$out/bin:$PATH
+    # Still don't know why this fails
+    sed -i 's/--set dir=hello //' tests/bash/test_default.sh
+
+    # This test has been disabled since it requires a network connaction
+    sed -i 's/test_downloader_getter(self):/disabled_test_downloader_getter(self):/' papis/downloaders/tests/test_main.py
+
     export HOME=$(pwd)/check-phase
+    make test
+    make test-non-pythonic
```
Comment (dotlambda) [inline pkgs/tools/misc/papis/default.nix]: `SH=${bashInteractive}/bin/bash make test-non-pythonic` should solve the problem.

**Comment by nico202** [comment]
@dotlambda thanks a lot, it seems to be ok now

**Comment by dotlambda** [comment]
@GrahamcOfBorg build papis

**Review by dotlambda — APPROVED** [review]
Thanks a lot!

**Pull request merged**